File: tests/typesafety/python/literal_flag_misuse.py
# expect: fails
# title: A variable typed str, rejected at the mode parameter
# misuse-of: literal_flag
from typing import Literal


def read_log(path: str, mode: Literal["text", "binary"]) -> int:
    record_size = 1 if mode == "text" else 8
    return len(path) * record_size


def tail(chosen: str) -> int:
    return read_log("app.log", chosen)  # error: str is wider than the two values

File: tests/typesafety/typescript/literal_flag_misuse.ts
// expect: fails

function readLog(path: string, mode: "text" | "binary"): number {
  const recordSize = mode === "text" ? 1 : 8;
  return path.length * recordSize;
}

export function tail(chosen: string): number {
  return readLog("app.log", chosen);
}
